### PR TITLE
Tp 1336 fix repeating activities

### PR DESCRIPTION
**Pull request by danijel**
Very hard to reproduce the issue. 
After playing with manually unsetting current activity I've managed to do it. Looks like there is a race condition between state update via cache and manual one.

As if when we did manual update we were not working with up to date activities array, which cause us to select same activity again.

With this PR determining what should current activity be is now driven solely by cache update, and activity can change only when actual status in activity array changes.

In addition to this polling for activities is no done on demand, only when we have no current activity. I've tested this with subscription disabled and it works fine.

Also I did a cleanup of timer we no longer need.

https://www.loom.com/share/8d479878a21044f6b65aa4f7046e2263?sid=f361fa77-c530-4ae3-b6cc-6206308ee2c9

### CODE REVIEW

**Comment by danijel** [comment]
@ebomcke-awell @bejoinka since you both watched the looms from the ticket and are familiar with this, can I get a review from one of you. Thanks.

File: src/components/Activities/context/ActivityProvider.tsx [hunk 1]
```diff
@@ -7,48 +7,46 @@ import { useTranslation } from 'next-i18next'
...
+      if (isNil(firstActive)) {
+        // nothing to activate, start polling for new activities so we don't rely on subscriptions
+        startPolling(POLLING_DELAY)
+      } else {
+        // we have something to activate, stop polling, no need for it
+        stopPolling()
       }
+      // set to the first active activity or undefined if none found,
+      // undefined will result in loading page with reload timer
+      setCurrentActivity(firstActive)
```
Comment (ebomcke-awell) [inline src/components/Activities/context/ActivityProvider.tsx:46]: This makes the intent clearer and remove the need to explain it via comments.

```suggestion
      if (isNil(firstActive)) {
        // nothing to activate, start polling for new activities so we don't rely on subscriptions
        setCurrentActivity(undefined)
        startPolling(POLLING_DELAY)
      } else {
        // we have something to activate, stop polling, no need for it
        setCurrentActivity(firstActive)
        stopPolling()
      }
```

Side note: is it safe to call `stopPolling` if there is no active polling ?

**Review by ebomcke-awell — APPROVED** [review]
Change is a suggestion so not blocking for the merge.

File: src/components/Activities/context/ActivityProvider.tsx (hunk 1 above)
Comment (danijel) [reply src/components/Activities/context/ActivityProvider.tsx]: yes, I tried, I'll confirm once again

**Pull request merged**